src/renderer/components/ExplorationStatus/ExplorationTimer.tsx

```diff
@@ -0,0 +1,123 @@
+import { Button } from '@/renderer/components/common/Button';
+import { rosClient } from '@/renderer/utils/ros/rosClient';
+import React, { useEffect, FC, useState, ChangeEvent } from 'react';
+import { LabeledInput } from '@/renderer/components/common/LabeledInput';
+import { log } from '@/renderer/logger';
+import { styled } from '@/renderer/globalStyles/styled';
+
+export const ExplorationTimer: FC = () => {
+  const [duration, setDuration] = useState(2);
+  const [timeRemaining, setTimeRemaining] = useState(Number);
+  const [isTimerActive, setIsTimerActive] = useState(false);
+  const [timerDisplay, setTimerDisplay] = useState('00:00');
+  const [countDownDate, setCountDownDate] = useState(Date.now());
+
+  const updateDuration = (e: ChangeEvent<HTMLInputElement>) => {
+    setDuration(Number(e.target.value));
+  };
+
+  const startTimer = () => {
+    setIsTimerActive(false);
+    setCountDownDate(Date.now() + duration * 60 * 1000);
+    setIsTimerActive(true);
+    setRosExplorationTimer();
+  };
+
+  const stopTimer = () => {
+    setIsTimerActive(false);
+    setDuration(0);
+    setRosExplorationTimer();
```

Comment:
Why call setRosExplorationTimer here?

Reply:
to set the ros /start_exploration to 0, so it stop